components/script/dom/bindings/typedarrays.rs

```diff
@@ -26,6 +26,35 @@ unsafe impl<T> crate::dom::bindings::trace::JSTraceable for HeapTypedArray<T> {
     }
 }
 
+pub fn new_initialized_heap_typed_array<T>(init: HeapTypedArrayInit) -> HeapTypedArray<T>
+where
+    T: TypedArrayElement + TypedArrayElementCreator,
+    T::Element: Clone + Copy,
+{
+    match init {
+        HeapTypedArrayInit::Object(js_object) => HeapTypedArray {
+            internal: Heap::boxed(js_object),
+            phantom: PhantomData::default(),
+        },
+        HeapTypedArrayInit::Info { len, cx } => {
+            rooted!(in (*cx) let mut array = ptr::null_mut::<JSObject>());
+            let typed_array =
+                create_typed_array_with_length::<T>(cx, len as usize, array.handle_mut())
+                    .expect("Creating TypedArray from length should never fail");
+            let heap_typed_array = HeapTypedArray::<T>::default();
+            heap_typed_array
+                .set_data(cx, unsafe { typed_array.as_slice() })
```

Comment:
unsafe is unnecessary, because the entire module allows it.

Reply:
I still get `this operation is unsafe and requires an unsafe function or block` when I remove it

Reply:
I thought it's better to keep `unsafe` here instead of using it in `new_with_jsobject`

Reply:
ah yes sorry, you need the unsafe block(but there is no need to mark the method as allowing unsafe, which is what I had in mind somehow).